File: LibraryAggregator.DataLayer/Entities/Booking.cs
using LibraryAggregator.DataLayer.Entities.Enum;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraryAggregator.DataLayer.Entities
{
    public class Booking
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int BookingId { get; set; }
        public int Code { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime FinishDate { get; set; }
        public int BooksLibraryId { get; set; }
        public BooksLibrary? BooksLibrary { get; set; }
        public int ClientId { get; set; }
        public virtual Client? Client { get; set; }
        public BookingStatuses BookingStatus
        {
            get; set;
        }
    }
}

File: LibraryAggregator.WebClient/src/app/models/booking.ts
import { BookingStatus } from "./bookingStatus";
import { BookLibrary } from "./bookLibrary";
import { Client } from "./client";

export interface Booking {
    bookingId?: number;
    code?: number;
    startDate: Date;
    finishDate: Date;
    clientId: number;
    client?: Client;
    booksLibraryId: number;
    booksLibrary?: BookLibrary;
    bookingStatusId?: number;
    bookingStatus?: BookingStatus;
}
